bucket4j-hazelcast-all/bucket4j-hazelcast/src/main/java/io/github/bucket4j/grid/hazelcast/serialization/SerializationUtilities.java

```diff
@@ -0,0 +1,84 @@
+package io.github.bucket4j.grid.hazelcast.serialization;
+
+import com.hazelcast.internal.util.StringUtil;
+import com.hazelcast.nio.serialization.Serializer;
+
+import java.text.MessageFormat;
+import java.util.*;
+import java.util.function.Supplier;
+
+public class SerializationUtilities {
+    public static final String TYPE_ID_BASE_PROP_NAME = "Bucket4jSerializers_TypeId_Base";
+    private static final Map<Class<? extends Serializer>, Integer> serializerTypeIdOffsets = Map.ofEntries(
+            new AbstractMap.SimpleEntry<Class<? extends Serializer>, Integer>(HazelcastEntryProcessorSerializer.class, 0),
+            new AbstractMap.SimpleEntry<Class<? extends Serializer>, Integer>(SimpleBackupProcessorSerializer.class, 1),
+            new AbstractMap.SimpleEntry<Class<? extends Serializer>, Integer>(HazelcastOffloadableEntryProcessorSerializer.class, 2)
+    );
+
+    public static int getSerializerTypeId(Class<? extends Serializer> serializerType) {
+        var typeIdBase = getSerializersTypeIdBase();
+
+        if (typeIdBase.isEmpty()) {
+            String msg = MessageFormat.format("Missing TypeIdBase number, impossible to load Bucket4j custom serializers. It must be provided in form of Environment Variable or System Property, both using the following key: [{0}]", TYPE_ID_BASE_PROP_NAME);
+            throw new MissingConfigurationParameterException(msg);
+        }
+
+        return getSerializerTypeId(serializerType, typeIdBase.get());
+    }
+
+    public static int getSerializerTypeId(Class<? extends Serializer> serializerType, int typeIdBase) {
+        return typeIdBase + SerializationUtilities.getSerializerTypeIdOffset(serializerType);
+    }
+
+    private static Optional<Integer> getSerializersTypeIdBase() {
+        return Optional.ofNullable(
+                getSerializerTypeIdBaseFromSystemProperty()
+                        .orElseGet(() -> getSerializerTypeIdBaseFromEnvironmentVariable()
+                                .orElseGet(() -> null)));
+    }
+
+    private static int getSerializerTypeIdOffset(Class<? extends Serializer> serializerType) {
+        if (serializerTypeIdOffsets.containsKey(serializerType)) {
+            return serializerTypeIdOffsets.get(serializerType);
+        } else {
+            String msg = MessageFormat.format("The internal configuration does not include any offset for the serializerType [{0}]", serializerType);
+            throw new IllegalStateException(msg);
+        }
+    }
+
+    private static Optional<Integer> getSerializerTypeIdBaseFromEnvironmentVariable() {
+        return getPropertyValueFromExternal(() -> System.getenv(SerializationUtilities.TYPE_ID_BASE_PROP_NAME), "Environment Variable");
```

Comment:
The generic rule that is most Java libraries obeys - is following:
* Firstly trying to get parameter from static variable. 
* If undefined, then getting parameters from `System.getProperty`
* If parameter still undefined then getting it from `System.getenv`

Reply:
I mean that it should be ability to calll something like this ```SerializationUtilities.setBaseTypeIdOffset(...)``` if user want. 

But I have rethinked this yet another time and made conclusion that it is overcomplication, because user already has ability to configure all that he want via ```HazelcastProxyManager.addCustomSerializers```, so I thing it is not neccesary to add functionality for configuration via static methods. Anyway it needs to add ability to configure baseId via System Properties, and  this way should have higher priority that System Env.

Reply:
I have missed that System Properties is already supported.

Reply:
> I mean that it should be ability to calll something like this `SerializationUtilities.setBaseTypeIdOffset(...)` if user want.

Now I got your point, u right, in a plain case it is a good idea to add flexibility.

> But I have rethinked this yet another time and made conclusion that it is overcomplication, because user already has ability to configure all that he want via `HazelcastProxyManager.addCustomSerializers`, so I thing it is not neccesary to add functionality for configuration via static methods. Anyway it needs to add ability to configure baseId via System Properties, and this way should have higher priority that System Env.

> I have missed that System Properties is already supported.

Exactly!